components/tidb_query_datatype/src/codec/mysql/json/mod.rs

```diff
@@ -203,6 +203,19 @@ impl<'a> JsonRef<'a> {
     pub(crate) fn get_str(&self) -> Result<&'a str> {
         Ok(str::from_utf8(self.get_str_bytes()?)?)
     }
+
+    // Return whether the value is zero.
+    pub(crate) fn is_zero(&self) -> bool {
+        match self.get_type() {
+            JsonType::Object => false,
+            JsonType::Array => false,
+            JsonType::Literal => false,
+            JsonType::I64 => self.get_i64() == 0,
+            JsonType::U64 => self.get_u64() == 0,
+            JsonType::Double => self.get_double() == 0f64,
+            JsonType::String => false,
```

Comment:
We don't need to check the json string's length here?

Reply:
I don't think so. I have a test in MySQL8.0.  
```
MySQL [test]> select *,json_extract(j, '$.test') from testjson where json_extract(j, '$.test');
+----+-----------------+---------------------------+
| id | j               | json_extract(j, '$.test') |
+----+-----------------+---------------------------+
|  1 | {"test": ""}    | ""                        |
|  2 | {"test": ""}    | ""                        |
|  3 | {"test": true}  | true                      |
|  4 | {"test": false} | false                     |
+----+-----------------+---------------------------+
4 rows in set, 1 warning (0.000 sec)
```

Reply:
Yes, the logic here is correctly, here always return `false` no matter the value is.

Maybe we can add ref: https://dev.mysql.com/doc/refman/8.0/en/json.html#Converting%20between%20JSON%20and%20non-JSON%20values here?

```
MariaDB [test]> CREATE TABLE JSON (c JSON);
MariaDB [test]> insert into json values row('{"a":"","b":"abc"}');
MariaDB [test]> select if(JSON_EXTRACT(c, '$.a') = true, 'TRUE', 'FALSE') from JSON;
+----------------------------------------------------+
| if(JSON_EXTRACT(c, '$.a') = true, 'TRUE', 'FALSE') |
+----------------------------------------------------+
| FALSE                                              |
+----------------------------------------------------+
1 row in set (0.001 sec)
MariaDB [test]> select if(JSON_EXTRACT(c, '$.b') = true, 'TRUE', 'FALSE') from JSON;
+----------------------------------------------------+
| if(JSON_EXTRACT(c, '$.b') = true, 'TRUE', 'FALSE') |
+----------------------------------------------------+
| FALSE                                              |
+----------------------------------------------------+
1 row in set (0.001 sec)
```